plugins/vpp/ifplugin/bfd_config.go

```diff
@@ -190,222 +192,234 @@ func (plugin *BFDConfigurator) ModifyBfdSession(oldBfdInput *bfd.SingleHopBFD_Se
+			err := c.bfdHandler.AddBfdUDPSession(bfdSession, ifIdx, c.keysIndexes)
 			if err != nil {
-				return err
+				return errors.Errorf("failed to add BFD UDP session for interface %s: %v",
+					bfdSession.Interface, err)
 			}
 		}
-		plugin.log.Debugf("%v session(s) recreated", len(sessionList.Session))
+		c.log.Debugf("%d session(s) recreated", len(sessionList.Session))
 	}
 	return nil
 }
 
 // ConfigureBfdEchoFunction is used to setup BFD Echo function on existing interface
-func (plugin *BFDConfigurator) ConfigureBfdEchoFunction(bfdInput *bfd.SingleHopBFD_EchoFunction) error {
-	plugin.log.Infof("Configuring BFD echo function for source interface %v", bfdInput.EchoSourceInterface)
-
+func (c *BFDConfigurator) ConfigureBfdEchoFunction(bfdInput *bfd.SingleHopBFD_EchoFunction) error {
 	// Verify interface presence
-	_, _, found := plugin.ifIndexes.LookupIdx(bfdInput.EchoSourceInterface)
+	_, _, found := c.ifIndexes.LookupIdx(bfdInput.EchoSourceInterface)
 	if !found {
-		return fmt.Errorf("interface %v does not exist", bfdInput.EchoSourceInterface)
+		return errors.Errorf("BFD echo function add: interface %s does not exist", bfdInput.EchoSourceInterface)
 	}
 
-	err := plugin.bfdHandler.AddBfdEchoFunction(bfdInput, plugin.ifIndexes)
+	err := c.bfdHandler.AddBfdEchoFunction(bfdInput, c.ifIndexes)
 	if err != nil {
-		return fmt.Errorf("error while setting up BFD echo source with interface %v", bfdInput.EchoSourceInterface)
+		return errors.Errorf("failed to set BFD echo source for interface %s: %v",
+			bfdInput.EchoSourceInterface, err)
 	}
 
-	plugin.echoFunctionIndex.RegisterName(bfdInput.EchoSourceInterface, plugin.bfdIDSeq, nil)
-	plugin.log.Debugf("BFD echo function with interface %v registered. Idx: %v", bfdInput.EchoSourceInterface, plugin.bfdIDSeq)
-	plugin.bfdIDSeq++
+	c.echoFunctionIndex.RegisterName(bfdInput.EchoSourceInterface, c.bfdIDSeq, nil)
+	c.log.Debugf("BFD echo function for interface %s registered", bfdInput.EchoSourceInterface)
+	c.bfdIDSeq++
 
-	plugin.log.Infof("Echo source set to interface %v ", bfdInput.EchoSourceInterface)
+	c.log.Infof("BFD echo source set for interface %s ", bfdInput.EchoSourceInterface)
 
 	return nil
 }
 
 // ModifyBfdEchoFunction handles echo function changes
-func (plugin *BFDConfigurator) ModifyBfdEchoFunction(oldInput *bfd.SingleHopBFD_EchoFunction, newInput *bfd.SingleHopBFD_EchoFunction) error {
-	plugin.log.Debug("There is nothing to modify for BFD echo function")
+func (c *BFDConfigurator) ModifyBfdEchoFunction(oldInput *bfd.SingleHopBFD_EchoFunction, newInput *bfd.SingleHopBFD_EchoFunction) error {
+	c.log.Warnf("There is nothing to modify for BFD echo function")
 	// NO-OP
+
+	/* todo: the reason is echo function uses interface name in key, so if interface is changed, the key changes (despite
+	   there is 'name' field in the model which is currently unused). Maybe it would be better to use name in the key,
```

Comment:
You meant `.. better to use name in the value`?

Reply:
yes, name from the value for registration, and also in the key (instead of interface name)